dm/syncer/syncer.go

```diff
@@ -3328,7 +3339,13 @@ func (s *Syncer) getEvent(tctx *tcontext.Context, startLocation binlog.Location)
 		return s.errOperatorHolder.GetEvent(startLocation)
 	}
 
-	return s.streamerController.GetEvent(tctx)
+	e, err := s.streamerController.GetEvent(tctx)
+	if err == nil {
+		s.locations.update(e)
+		// TODO: observe integration test
```

Comment:
will this implemented in this PR?

Reply:
what do you mean?

Reply:
> // TODO: observe integration test
when will this be done ?

Reply:
this todo means check the log printed by next line to see if locations are correct